testgrid/cmd/transfigure/cmd/transfigure.go

```diff
@@ -0,0 +1,363 @@
+			return err
+		}
+	}
+
+	if gitUser == "" || gitEmail == "" {
+		err = populateGitUserAndEmail()
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Print("Using user from Flag: " + gitUser)
+		log.Print("Using email from Flag: " + gitEmail)
+	}
+
+	err = createAndCheckoutGitBranch()
+	if err != nil {
+		return err
+	}
+
+	createRepoSubdirForTestgridYAML()
+
+	err = generateTestgridYAML()
+	if err != nil {
+		return err
+	}
+
+	err = gitAddAll()
+	if err != nil {
+		return err
+	}
+
+	if !gitDiffExists() {
+		log.Print("Transfigure did not change anything. Aborting no-op bump.")
+		return nil
+	}
+
+	err = runBazelTests()
+	if err != nil {
+		return err
+	}
+
+	if dryRun {
+		log.Print("Dry-Run; skipping PR")
+		return nil
+	}
+
+	err = gitCommitAndPush()
+	if err != nil {
+		return err
+	}
+
+	err = createPR()
+	return err
+}
+
+func init() {
+	// Required Flags
+	rootCmd.PersistentFlags().StringVar(&prowConfig, "prow_config", "", "Config path for your non-k8s Prow instance.")
+	rootCmd.PersistentFlags().StringVar(&prowJobConfig, "prow_job_config", "", "Job Config path for your non-k8s Prow instance.")
+	rootCmd.PersistentFlags().StringVar(&testgridYaml, "testgrid_yaml", "", "TestGrid configuration directory or default file.")
+	rootCmd.PersistentFlags().StringVar(&repoSubdir, "repo_subdir", "", "The subdirectory in /config/testgrids/... to push to. Usually <github_org> or <github_org>/<github_repository>.")
+	for _, flag := range []string{"prow_config", "prow_job_config", "testgrid_yaml", "repo_subdir"} {
+		rootCmd.MarkPersistentFlagRequired(flag)
+	}
+
+	// Optional Flags
+	rootCmd.PersistentFlags().StringVar(&githubToken, "github_token", "", "A GitHub personal access token. A value of 'Test' will trigger a dry run.")
+	rootCmd.PersistentFlags().StringVar(&remoteForkRepo, "remote_fork_repo", "", "The name of the user's fork of kubernetes/test-infra, if it is not named test-infra.")
+	rootCmd.PersistentFlags().StringVar(&gitUser, "git_user", "", "The GitHub username to use. Pulled from github_token if not specified.")
+	rootCmd.PersistentFlags().StringVar(&gitEmail, "git_email", "", "The GitHub email to use. Pulled from github_token if not specified.")
+	rootCmd.PersistentFlags().BoolVar(&dryRun, "dry_run", false, "If true, no PR will be created.")
+}
+
+func createTempWorkingDir() error {
+	tempDir, err := ioutil.TempDir(".", "transfigure_")
+	if err != nil {
+		return wrapErrorOrNil("Error creating temp dir", err)
+	}
+	workingDir, err = filepath.Abs(tempDir)
+	if err != nil {
+		return wrapErrorOrNil("Error getting abs path for working dir", err)
+	}
+	log.Printf("Created temp directory %v/", tempDir)
+	return nil
+}
+
+func removeTempWorkingDir() {
+	os.RemoveAll(workingDir)
+	log.Printf("Removed temp directory %v/", workingDir)
+}
+
+func cloneK8sTestInfraRepo() error {
+	log.Printf("Cloning kubernetes/test-infra into temp directory")
+	cmd := exec.Command("git", "clone", "https://github.com/kubernetes/test-infra.git")
```

Comment:
Can we make this value `https://github.com/kubernetes/test-infra.gits` as const?